lib/noticed/delivery_methods/one_signal.rb

```diff
@@ -0,0 +1,69 @@
+module Noticed
+  module DeliveryMethods
+    class OneSignal < DeliveryMethod
+      required_options :target_channel
+
+      def deliver
+        response = post_request(url, headers: headers, json: json)
+        parsed_response = JSON.parse(response.body)
+
+        # OneSignal might returns 200 with errors
+        # - "All included devices are not subscribed"
+        # - Invalid aliases
+        if parsed_response.key?("errors")
+          raise ResponseUnsuccessful.new(response, url, headers: headers, json: json)
+        end
+      rescue Noticed::ResponseUnsuccessful => exception
+        if config[:error_handler]
+          notification.instance_exec(exception.response, &config[:error_handler])
+        else
+          raise
+        end
+      end
+
+      def json
+        evaluate_option(:json).merge({
```

Comment:
This will override the given JSON, so it's probably better to use this?

```suggestion
        evaluate_option(:json).with_defaults({
```

I think the `||` is also unnecessary since won't you always have an object so the `or` would never evaluate?